internal/bioscfg/bioscfg.go

```diff
@@ -0,0 +1,113 @@
+package bioscfg
+
+import (
+	"context"
+
+	"github.com/metal-toolbox/bioscfg/internal/config"
+	"github.com/metal-toolbox/bioscfg/internal/store/fleetdb"
+	"github.com/metal-toolbox/ctrl"
+	rctypes "github.com/metal-toolbox/rivets/condition"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	pkgName = "internal/bioscfg"
+)
+
+// BiosCfg BiosCfg Controller Struct
+type BiosCfg struct {
+	cfg     *config.Configuration
+	logger  *logrus.Entry
```

Comment:
I thought we were moving away from `logrus`?

Also, I really don't like having to pass around the logger. It really limits your ability to log, unless you include the logger everywhere, which then bloats the code because you have to have the logger as a field on every struct and include it in every `new()` method.

Reply:
I vageuly remember us talking about it, but we havent really done it anywhere else. I am 100% down to remove thirdparty dependencies and move to slog. I will take a look.

Reply:
After talking with John, we are deciding to postpone the work to move over to logrus until a later date